helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/Jackson1Utils.java

```diff
@@ -61,4 +70,43 @@ public static JsonNode toJsonNode(Object datum) {
     }
   }
 
+  // See the doc for AvroCompatibilityHelper.setFieldPropFromJsonString() for how str and strict are interpreted.
+  public static JsonNode toJsonNode(String str, boolean strict) {
+    if (str == null) {
+      return null;
+    }
+    try {
+      JsonParser parser = JSON_FACTORY.createJsonParser(str);
+      JsonNode node = OBJECT_MAPPER.readTree(parser);
+      if (node == null) {
+        // Happens when str.isEmpty().
+        throw new NullPointerException("str cannot be an empty string");
+      }
+      // Calling parser.nextToken() can raise an exception itself, if there are garbage chars at the end.
+      // E.g., when str == "1,2", the ',' is the garbage char (raises JsonParseException).
+      if (parser.nextToken() != null) {
+        // Or, if that succeeds and returns a valid token, the extra chars (while not garbage) are still unwanted.
+        // E.g., when str == "1 2", the '2' is the extra token.
+        throw new IllegalStateException("str cannot have extra trailing chars; must be a single literal");
```

Comment:
could you include location (from parser.getTokenLocation()) and maybe the "junk string" ? (the part of the string remaining past the location where parsing stopped)

Reply:
Good suggestion! Done.